src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/SyndicationFeed.cs

```diff
@@ -265,6 +275,300 @@ public TextSyndicationContent Title
             set { _title = value; }
         }
 
+        internal SyndicationLink InternalDocumentation
+        {
+            get
+            {
+                return _documentation;
+            }
+        }
+
+        public SyndicationLink Documentation
+        {
+            get
+            {
+                if (_documentation == null)
+                {
+                    _documentation = TryReadDocumentationFromExtension(ElementExtensions);
+                }
+
+                return _documentation;
+            }
+            set
+            {
+                _documentation = value;
+            }
+        }
+
+        internal int? InternalTimeToLive
+        {
+            get
+            {
+                return _timeToLive;
+            }
+        }
+
+        public int TimeToLive
+        {
+            get
+            {
+                if (_timeToLive == null)
+                {
+                    _timeToLive = TryReadTimeToLiveFromExtension(ElementExtensions);
+                }
+
+                return _timeToLive.Value;
```

Comment:
Is `InvalidOperationException` if `TryReadTimeToLiveFromExtension` returns null appropriate?

Reply:
Good catch. I changed the type of the property to `int?`. Thanks.